src/gen_module.sh

```diff
@@ -12,7 +12,7 @@ fi
 
 source_dir=${1}
 blender_dir=${2}
-branch_name=${3}
+tag_name=${3}
```

Comment:
I sometimes used this script to generate modules from specific commits or branches (such as master branch).
So, it should not only target to tags.

Reply:
To combine the shallow clone with your request, there a few options:
1. Hardcode check for master: If "master" is being used, checkout master branch
2. Try and error: If tag cannot be found, check for branch, if branch cannot be found, check for commit

Are those options OK with you?

Reply:
Found a better alternative. Fixed.